core/src/main/scala/kafka/server/LeaderEndPoint.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.server
+
+import kafka.api.{KAFKA_0_10_1_IV2, KAFKA_0_11_0_IV0, KAFKA_2_0_IV0, KAFKA_2_0_IV1, KAFKA_2_1_IV1, KAFKA_2_2_IV1, KAFKA_2_3_IV1, KAFKA_2_8_IV0, KAFKA_3_0_IV1}
+import kafka.utils.Logging
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.requests.FetchRequest
+import org.apache.kafka.common.message.OffsetForLeaderEpochResponseData.EpochEndOffset
+import org.apache.kafka.common.message.{FetchResponseData, OffsetForLeaderEpochRequestData}
+
+import scala.collection.Map
+
+trait LeaderEndPoint extends Logging {
+  val brokerConfig: KafkaConfig
```

Comment:
I also wonder about the config. The *Threads have also a reference to the KafkaConfig so I don't really see the value of providing it via the interface as well. What was the reasoning?

Reply:
Yeah, I think only `RemoteLeaderEndPoint` uses the `brokerConfig` and the `*RequestVersion` functions. It might make more sense to optionally pass in the `brokerConfig` and the `*RequestVersion` to the functions that use it in `RemoteLeaderEndPoint` (i.e., all of the `fetch*Offsets` functions)

Reply:
Passing it in the constructor of `RemoteLeaderEndPoint` is fine. I just think that providing it via the interface to other components is weird.

Reply:
I agree. We only need to define the behavior (i.e. methods) in the interface.